pkg/internal/util/kustomize/kutomize_test.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kustomize
+
+import (
+	"testing"
+
+	"sigs.k8s.io/kind/pkg/internal/apis/config"
+	"sigs.k8s.io/kind/pkg/internal/util/assert"
+)
+
+func TestBuild(t *testing.T) {
+	type testCase struct{
+		Name string
+		Resources []string
+		Patches []string
+		PatchesJSON6902 []config.PatchJSON6902
+		ExpectError bool
+		ExpectOutput string
+	}
+	cases := []testCase{
```

Comment:
these are certainly not sufficient, however some is better than none.

next up I'm going to start tracking our unit test coverage in CI, but I wanted to have a package to test it out with, so that will be this package.

if everything works we'll see that the coverage is subpar, and then after a follow-up it will go up.